# zTest for Bus Route

Idea: The new bus route will be faster than the old route.

Null Hypothesis: The average duration of trips on the new route will be greater than or equal to the average duration of trips on the old route with a significance level of 0.05.

Alt Hypothesis: The average duration of trips on the new route will be significantly less than the average of trips on the old route with a significance value of 0.05.

In [2]:
#Imports
import pandas as pd
import numpy as np
import matplotlib as plt
%pylab inline


Populating the interactive namespace from numpy and matplotlib


In [20]:
#New Bus Route
data = pd.read_table("https://raw.githubusercontent.com/fedhere/PUI2017_fb55/master/Lab3_fb55/times.txt", header = None).values.squeeze()
data

array([ 31.62223931,  32.82137636,  30.2291014 ,  31.41376587,
        39.01055035,  34.82207891,  39.87188366,  39.57994562,
        31.02658678,  27.66246068,  18.8752884 ,  46.2313731 ,
        30.01522604,  47.6004104 ,  16.52726666,  42.4462331 ,
        33.62527449,  42.28462558,  46.20520667,  30.87517392,
        32.94800341,  26.17691512,  31.25699649,  35.33644554,
        35.64072501,  40.32297174,  38.04348588,  29.4579793 ,
        23.91636367,  34.56403521,  45.42782464,  38.95607754,
        39.11396748,  39.67398759,  31.89148105,  22.62379325,
        26.18910408,  43.54114213,  33.2607386 ,  31.67602384,
        37.12846259,  38.22048496,  34.38814185,  36.11407371,
        44.08064814,  29.59917161,  37.02411751,  41.60576556,
        44.37860528,  38.06471077,  22.90526456,  40.27182857,
        31.57979239,  25.88037795,  28.8913048 ,  44.33687925,
        36.78948684,  43.46098875,  40.65557   ,  26.16634004,
        25.13884784,  33.01581456,  32.87050073,  26.28

In [22]:
#Calc new mean
newMean = data.mean()
newMean

34.466161688299998

In [23]:
#Calc new standard deviation
newStd = data.std()
newStd

7.1015040681937771

In [24]:
#load in old (or 'pop') mean & standard deviation
popMean = 36
popStd = 6

In [29]:
#Calc sqrt of N 
sqrt_N = np.sqrt(len(data))

In [31]:
#Run Z Test
((popMean - newMean) / (popStd / sqrt_N))

2.5563971861666701

# Analysis:
Since Z test resulted in a value of 2.56, which is greater than 2, we can say that the new bus route is faster than the old bus route.